L1Trigger/TrackTrigger/src/HitPatternHelper.cc

```diff
@@ -15,11 +15,59 @@ namespace hph {
 
   Setup::Setup(const edm::ParameterSet& iConfig, const tt::Setup& setupTT)
       : iConfig_(iConfig),
+        oldKFPSet_(iConfig.getParameter<edm::ParameterSet>("oldKFPSet")),
         setupTT_(setupTT),
+        hphDebug_(iConfig.getParameter<bool>("hphDebug")),
+        useNewKF_(iConfig.getParameter<bool>("useNewKF")),
+        deltaTanL_(iConfig.getParameter<double>("deltaTanL")),
+        cotNbins_(iConfig.getParameter<int>("cotNbins")),
+        z0Nbins_(iConfig.getParameter<int>("z0Nbins")),
+        chosenRofZ_(),
+        beamWindowZ_(),
+        cotMax_(),
+        etaRegions_(),
         layerIds_(),
         layermap_(),
+        layerEncoding_(cotNbins_, std::vector<std::vector<tt::SensorModule>>(z0Nbins_)),
         nEtaRegions_(tmtt::KFbase::nEta_ / 2),
         nKalmanLayers_(tmtt::KFbase::nKFlayer_) {
+    if (useNewKF_) {
+      chosenRofZ_ = setupTT_.chosenRofZ();
+      beamWindowZ_ = setupTT_.beamWindowZ();
+      etaRegions_ = setupTT_.boundarieEta();
+    } else {
+      chosenRofZ_ = oldKFPSet_.getParameter<double>("ChosenRofZ");
+      beamWindowZ_ = oldKFPSet_.getParameter<double>("BeamWindowZ");
+      etaRegions_ = oldKFPSet_.getParameter<vector<double>>("EtaRegions");
+    }
+    double etaMax_ = abs(etaRegions_[0]);
+    cotMax_ = abs(1 / tan(2. * atan(exp(-etaMax_))));
+    for (int cotBin = 0; cotBin < cotNbins_; cotBin++) {
+      double cotDigi_ = (2 * cotMax_ / cotNbins_) * (cotBin + 1) - cotMax_;
+      for (int z0Bin = 0; z0Bin < z0Nbins_; z0Bin++) {
+        double z0Digi_ = (2 * beamWindowZ_ / z0Nbins_) * (z0Bin + 1) - beamWindowZ_;
+        std::vector<tt::SensorModule>& layers_ = layerEncoding_[cotBin][z0Bin];
+        //Looping over sensor modules to make predictions on which layers particles are expected to hit
+        for (const tt::SensorModule& sm : setupTT_.sensorModules()) {
+          double d = (z0Digi_ - sm.z() + sm.r() * cotDigi_) / (sm.cosTilt() - sm.sinTilt() * cotDigi_);
+          double d_p = (z0Digi_ - sm.z() + sm.r() * (cotDigi_ + deltaTanL_ / 2)) /
+                       (sm.cosTilt() - sm.sinTilt() * (cotDigi_ + deltaTanL_ / 2));
+          double d_m = (z0Digi_ - sm.z() + sm.r() * (cotDigi_ - deltaTanL_ / 2)) /
+                       (sm.cosTilt() - sm.sinTilt() * (cotDigi_ - deltaTanL_ / 2));
+          if (useNewKF_ &&
```

Comment:
Why is the logic here different for assigning a SensorModule to an (eta,z0) sector depending if NewKF is used or not?

Reply:
Done, the logic is the same for old and new kf now